### PR TITLE
recursively create config directory

### LINKED ISSUE #28: Recursive-Create configuration directory
Why don't we do this? This is a problem.

### PR DESCRIPTION
**Pull request by tschuy**
fixes #28

### CODE REVIEW

**Comment by tschuy** [comment]
Unit tests, style tests, and so far the one completed integration test run have all passed successfully locally.

**Comment by tschuy** [comment]
@jordane in lieu of being able to @osuosl-cookbooks/chefs :crying_cat_face:

File: providers/default.rb [hunk 1]
```diff
@@ -82,6 +82,15 @@
     group new_resource.group
   end
 
+  # Create config directory if needed.
+  config_directory = ::File.dirname(config_destination)
+  directory config_directory do
+    only_if { !new_resource.config_template.nil? }
```
Comment (jordane) [inline providers/default.rb]: There exists a `not_if` for exactly this purpose.

**Comment by jordane** [comment]
+1

**Pull request merged**